### PR TITLE
Document `vaccinations` table

### LINKED ISSUE #1340: Improve table schemas documentation
The [schema docs](https://docs.opensafely.org/ehrql/reference/schemas/beta.tpp/) should tell users all that we know about the data in each table, including:

* what it contains
* where it comes from
* warnings about known issues with the data
* whether it is refreshed regularly (and if not, we should link to the database history report)
* links to any external documentation
* links to any short data reports
* a description of the contents of each column

Documentation can be cribbed from [existing data sources](https://docs.opensafely.org/data-sources/) docs, but we may have to search Slack, GH, and the cohort-extractor codebase for other information.#

This is also the opportunity to consider whether tables and columns are correctly named.

We'll divide up the work across the whole data team, and with issues for each table (or group of tables).

- [ ] #1371 
- [x] #1372
- [x] #1373
- [ ] #1374
- [ ] #1375
- [ ] #1376
- [ ] #1377
- [x] #1378
- [ ] #1379
- [x] #1380
- [x] #1381
- [x] #1382
- [ ] #1383
- [ ] #1384
- [ ] #1385

### PR DESCRIPTION
**Pull request by milanwiedemann**
Improve documentation of the `vaccinations` table. In our current documentation we only refer to this table in the [OpenSAFELY-TPP database](https://docs.opensafely.org/data-sources/systmone/#the-opensafely-tpp-database) section.

This adds a description for all columns and adds some context about vaccinations that might be missing from this table. I also included a link to a released output of all vaccination names, @iaindillingham suggested publishing this output on https://reports.opensafely.org/. A draft report already exists for this output but it is not yet available publicly and was last updated on 13 Dec 2022.

I'm waiting for more feedback from researchers before converting from draft to ready for review.

Closes #1340

### CODE REVIEW

**Comment by iaindillingham** [comment]
> I also included a link to a released output of all vaccination names, @iaindillingham suggested publishing this output on https://reports.opensafely.org/. A draft report already exists for this output but it is not yet available publicly and was last updated on 13 Dec 2022.

I don't understand this section, Milan. I think:

* An output has been released and published on OS Jobs: <https://jobs.opensafely.org/opensafely-internal/tpp-vaccination-names/outputs/25/output/vaccination_names.html>.
* This output has also been published on OS Reports, as a draft: <https://reports.opensafely.org/reports/opensafely-tpp-vaccination-names/>.
* Linking to an output on OS Reports, rather than on OS Jobs, is more consistent with other times we have linked to outputs in ehrQL's docs.

Why don't we publish the output on OS Reports and link to that? They're the same output and both are already on the Web.

**Comment by milanwiedemann** [comment]
> Why don't we publish the output on OS Reports and link to that? They're the same output and both are already on the Web.

I agree, we should get this released on OS reports. What I was trying to say is that the current report might be out of date (last updated more than 1 year ago) and the link to the _(Draft)_ on OS reports is not publicly available yet (I think).

**Comment by wjchulme** [comment]
I think it would be useful to be very clear that to identify (some) vaccination events in tpp you don't use snomed codes, but tpp-defined product names / target diseases. It might also be helpful to understand a bit more about why that is, but I don't know. I think perhaps @brianmackenna does?

I don't think this should hold up the merge, but maybe capture in an issue

File: ehrql/tables/beta/tpp.py
Comment (iaindillingham) [inline ehrql/tables/beta/tpp.py:633]: Don't forget to check [the rendered docs](https://milanwiedemann-document-vacc.databuilder.pages.dev/reference/schemas/beta.tpp/#vaccinations). This "Vaccinations" doesn't contribute anything.

```suggestion
```

**Review by iaindillingham — APPROVED** [review]
I've approved, but incorporating my suggestion before merging would make me happy 🙂

**Pull request merged**
